xmrstak/backend/amd/amd_gpu/opencl/cryptonight.cl

```diff
@@ -794,6 +795,17 @@ __kernel void JOIN(cn1,ALGO) (__global uint4 *Scratchpad, __global ulong *states
 		}
 #endif
 
+#if(ALGO == cryptonight_v8_reversewaltz)
```

Comment:
One slight cleanup here:
```C++
#endif

#if(ALGO ==
```
would be better written as:
```C++
#elif(ALGO ==
```

Reply:
Makes sense. Thanks.
Done.

Reply:
Actually I think this code could be made simpler still if you did some `#define` trickery on the declaration, like this:

```C++
// (earlier)
#if(ALGO == cryptonight_monero_v8)
#  define CNV8_CHUNK1_INTO chunk1
#  define CNV8_CHUNK3_INTO chunk3
#elif(ALGO == cryptonight_v8_reversewaltz)
#  define CNV8_CHUNK1_INTO chunk3
#  define CNV8_CHUNK3_INTO chunk1
#endif
```

Then you can ditch the second code block entirely and change the first to:
```C++
#if(ALGO == cryptonight_monero_v8 || ALGO == cryptonight_v8_reversewaltz)
		{
			ulong2 CNV8_CHUNK1_INTO = as_ulong2(SCRATCHPAD_CHUNK(1));
			ulong2 chunk2 = as_ulong2(SCRATCHPAD_CHUNK(2));
			ulong2 CNV8_CHUNK3_INTO = as_ulong2(SCRATCHPAD_CHUNK(3));
			SCRATCHPAD_CHUNK(1) = as_uint4(chunk3 + ((ulong2 *)(b_x + 1))[0]);
			SCRATCHPAD_CHUNK(2) = as_uint4(chunk1 + ((ulong2 *)b_x)[0]);
			SCRATCHPAD_CHUNK(3) = as_uint4(chunk2 + ((ulong2 *)a)[0]);
		}
#endif

and then... (continued in next comment)

Reply:
Yes, but it can complicate reading of the code. Also, I wanted to minimize my changes, so I tried to leave original implementation as close to original as I could. However, if @psychocrypt and @fireice-uk agree with your proposal, I'll replace it using defines.